tests/test_scheduler.py

```diff
@@ -0,0 +1,110 @@
+"""Unit tests for monitor module."""
+
+import unittest
+
+from PyQt5.QtWidgets import QApplication
+from PyQt5.QtCore import Qt, QObject,  QMimeData
+
+from iquip.apps import scheduler
+from iquip.protocols import ExperimentInfo as ExpInfo
+
+class TestExperimentModel(unittest.TestCase):
+    """Unit tests for ExperimentModel class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_model_index(self):
+        data1 = []
+        for i in range(10):
+            data1.append(ExpInfo("exp" + str(i), {"rid": i, "priority": i}))
+        data2 = []
+        for i in range(100):
+            data1.append(ExpInfo("exp" + str(i), {"rid": i, "priority": 0}))
+        for data in (data1, data2):
+            mdl = scheduler.ExperimentModel()
+            mdl.experimentQueue.extend(data)
+            self.assertEqual(mdl.rowCount(), len(data))
+            for i, exp in enumerate(data):
+                self.assertEqual(mdl.data(mdl.index(i)), exp)
+
+    def test_drag_and_drop(self):
+        mdl = scheduler.ExperimentModel()
+        data = [ExpInfo("exp1", {"rid": 1, "priority": 2}),
+                ExpInfo("exp2", {"rid": 2, "priority": 1}),
+                ExpInfo("exp3", {"rid": 3, "priority": 1})
+               ]
+        mdl.experimentQueue.extend(data)
+        mime0 = QMimeData()
+        mime0.setText("0")
+        mime1 = QMimeData()
+        mime1.setText("1")
+        mime2 = QMimeData()
+        mime2.setText("2")
+        mdl.dropMimeData(mime0, Qt.MoveAction, 0, 0, mdl.index(0)) # exp1 above exp1
+        self.assertEqual(mdl.experimentQueue, data)
+        mdl.dropMimeData(mime0, Qt.MoveAction, 2, 0, mdl.index(0)) # exp1 above exp3
+        self.assertEqual(mdl.experimentQueue, data)
+        mdl.dropMimeData(mime1, Qt.MoveAction, 3, 0, mdl.index(0)) # exp2 below exp3
+        self.assertEqual(mdl.experimentQueue, [data[0], data[2], data[1]])
+        mdl.dropMimeData(mime2, Qt.MoveAction, 1, 0, mdl.index(0)) # exp2 above exp3
+        self.assertEqual(mdl.experimentQueue, data)
+
+
+class TestSchedulerApp(unittest.TestCase):
+    """Unit tests for SchedulerApp class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_add_experiment(self):
+        app = scheduler.SchedulerApp(name="name", parent=QObject())
+        data = [ExpInfo("exp" + str(i), {"rid": i, "priority": 10-i}) for i in range(10)]
```

Comment:
Please rename "exp" and add spacings before and after `-` operator, i.e., `10 - i`.

Reply:
I thought `arginfo` contains RID and priority. Am I wrong?

Reply:
Oh, sorry for the late review..

When I defined `ExperimentInfo`, I thought this is a container of information about an experiment file, not a submitted experiment, so it has only the experiment and the build arguments.

But you used this in `scheduler.py` for describing a submitted experiment, so I think we should determine how to use this class.

Reply:
In my opinion, it looks better to define `SubmittedExperimentInfo` class separately in `protocols.py` and use this. @giwon2004

Reply:
Okay, I will define a new type and make a pull request for it.

Reply:
@BECATRUE Then should we merge this PR after the new PR (`SubmittedExperimentInfo`), or can we merge this now?